src/csharp/Grpc.Tools/ProtoCompile.cs

```diff
@@ -136,7 +136,7 @@ public class ProtoCompile : ToolTask
             new ErrorListFilter
             {
                 Pattern = new Regex(
-                    pattern: "(?'FILENAME'.+)\\((?'LINE'\\d+)\\) ?: ?warning in column=(?'COLUMN'\\d+) ?: ?(?'TEXT'.*)",
+                    pattern: "^(?'FILENAME'.+?)\\((?'LINE'\\d+)\\) ?: ?warning in column=(?'COLUMN'\\d+) ?: ?(?'TEXT'.*)",
```

Comment:
I think one could still come up with a counterexample, but it's basically impossible to accommodate all the corner cases. This shall work in 99.9% of use cases and should be good enough.